core/src/main/java/io/questdb/jit/CompiledFilterIRSerializer.java

```diff
@@ -871,8 +872,12 @@ private void serializeNumber(
     }
 
     private void serializeOperator(int position, final CharSequence token, int argCount, int type) throws SqlException {
-        if (SqlKeywords.isInKeyword(token) && type == ExpressionNode.FUNCTION) {
-            serializeIn();
+        if (SqlKeywords.isInKeyword(token)) {
+            if (type == ExpressionNode.FUNCTION) {
+                serializeIn();
+            } else if (type == ExpressionNode.SET_OPERATION && isInTimestampPredicate()) {
+                serializeInTimestampRange(position);
+            }
             return;
```

Comment:
`return;` should be moved into above if branches. That's because we want to fall to the subsequent `throw` if there was no above match.

Reply:
Ah, true, thanks for the catch.